### ISSUE TITLE
How to implement the ‘enc_layer’？

### ISSUE THREAD

**Issue by gaunzi** [AsahiLiu/PointDetectron#1]
Hi，
I have some confusions about the implement of  ‘**enc_layer’** in **enc_layer.py**. 
The code of  **enc_layer’**  is : 

def enc_layer(D:int,K:int):
    net = nn.Sequential(Encoding(D, K),
                        # encoding.nn.EncodingV3(D=out_channel,K=K),
                        nn.ReLU(inplace=True),
                        #nn.ReLU(),
                        encoding.nn.View(-1, D * K),
                        encoding.nn.Normalize(),
                        nn.Linear(D * K, D),
                        nn.Sigmoid()
                        )
    return net

How could I find out the source code of  **'Encoding(D, K)'**?
Thanks!

**Comment by AsahiLiu** [comment]
> Hi，
> I have some confusions about the implement of ‘**enc_layer’** in **enc_layer.py**.
> The code of **enc_layer’** is :
> 
> def enc_layer(D:int,K:int):
> net = nn.Sequential(Encoding(D, K),
> # encoding.nn.EncodingV3(D=out_channel,K=K),
> nn.ReLU(inplace=True),
> #nn.ReLU(),
> encoding.nn.View(-1, D * K),
> encoding.nn.Normalize(),
> nn.Linear(D * K, D),
> nn.Sigmoid()
> )
> return net
> 
> How could I find out the source code of **'Encoding(D, K)'**?
> Thanks!



> Hi，
> I have some confusions about the implement of ‘**enc_layer’** in **enc_layer.py**.
> The code of **enc_layer’** is :
> 
> def enc_layer(D:int,K:int):
> net = nn.Sequential(Encoding(D, K),
> # encoding.nn.EncodingV3(D=out_channel,K=K),
> nn.ReLU(inplace=True),
> #nn.ReLU(),
> encoding.nn.View(-1, D * K),
> encoding.nn.Normalize(),
> nn.Linear(D * K, D),
> nn.Sigmoid()
> )
> return net
> 
> How could I find out the source code of **'Encoding(D, K)'**?
> Thanks!

Please refer to the source code and installation instructions in this link 

https://hangzhang.org/PyTorch-Encoding/notes/compile.html

After you finish the installation of encoding layer, it should be fine.

Best Regards,
